apps/omg_watcher/test/omg_watcher/utxo_exit/core_test.exs

```diff
@@ -25,82 +25,85 @@ defmodule OMG.Watcher.UtxoExit.CoreTest do
+                  {Utxo.Position.to_db_key(position),
+                   Utxo.to_db_value(%Utxo{amount: 10, currency: @eth, owner: alice.addr})}}
+               )
 
-  @tag fixtures: [:alice]
-  test "compose output exit", %{alice: alice} do
-    tx_encode = fn number ->
-      TestHelper.create_encoded([{1_000, number, 0, alice}], @eth, [{alice, 10}])
+      assert [%{amount: 10}] = txbytes |> Transaction.decode!() |> Transaction.get_outputs()
+      assert byte_size(proof) == 32 * 16
     end
 
-    tx_exit = tx_encode.(30)
-    tx_exit_raw_tx_bytes = Transaction.Signed.decode!(tx_exit) |> Transaction.raw_txbytes()
-
-    position = Utxo.position(3, 0, 1)
-    encode_utxo = position |> Utxo.Position.encode()
-
-    assert {:ok,
-            %{
-              proof: _proof,
-              sigs: _sig,
-              txbytes: ^tx_exit_raw_tx_bytes,
-              utxo_pos: ^encode_utxo
-            }} =
-             Core.compose_output_exit(
-               [
-                 %{txindex: 2, txbytes: tx_encode.(20)},
-                 %{txindex: 1, txbytes: tx_encode.(10)},
-                 %{txindex: 0, txbytes: tx_exit},
-                 %{txindex: 3, txbytes: tx_encode.(0)}
-               ],
-               position
-             )
+    test "fails to create deposit exit when UTXO missing in DB" do
+      assert {:error, :no_deposit_for_given_blknum} = Core.compose_deposit_standard_exit(:not_found)
+    end
   end
 
-  test "getting exit data returns error when there is no utxo" do
-    assert {:error, :utxo_not_found} == Core.compose_output_exit([], Utxo.position(1_000, 1, 2))
-  end
+  describe "compose_utxo_exit/2" do
+    test "composes output exit from tx inside a block", %{alice: alice} do
+      blknum = 4000
+      tx_exit = TestHelper.create_recovered([{1_000, 1, 0, alice}], @eth, [{alice, 10}])
+      tx_exit_raw_tx_bytes = Transaction.raw_txbytes(tx_exit)
+      position = Utxo.position(blknum, 1, 0)
+      encode_utxo = position |> Utxo.Position.encode()
 
-  @tag fixtures: [:alice]
-  test "return utxo when in blknum in utxos map", %{alice: %{addr: alice_addr}} do
-    assert %{amount: 7, currency: @eth, owner: alice_addr} ==
-             Core.get_deposit_utxo(
-               {:ok,
-                Map.new(Enum.map(1..20, fn a -> {{a, 0, 0}, %{amount: a, currency: @eth, owner: alice_addr}} end))},
-               Utxo.position(7, 0, 0)
-             )
-  end
+      block =
+        [
+          TestHelper.create_recovered([{1_000, 2, 0, alice}], @eth, [{alice, 10}]),
+          tx_exit,
+          TestHelper.create_recovered([{1_000, 3, 0, alice}], @eth, [{alice, 10}])
+        ]
+        |> Block.hashed_txs_at(blknum)
+        |> Block.to_db_value()
+
+      assert {:ok,
+              %{
+                proof: proof,
+                txbytes: ^tx_exit_raw_tx_bytes,
+                utxo_pos: ^encode_utxo
+              }} = Core.compose_block_standard_exit(block, position)
+
+      assert byte_size(proof) == 32 * 16
```

Comment:
`32 * 16` - doesn't tell me anything :open_hands: , also `64 * 8` would be misleading as there are bytes not bits...
I'm opting to make it local var, e.g. `expected_signature_byte_size = 512`

Reply:
it's 32 bytes times 16 (tree depth). It's the size of thep proof not signature. Also it's consistent with other occurrences of a similar test.

Not sure if adding an `expected_proof_byte_size = 32 * 16` is worth the extra line, this is exactly what the RHS of the assertion should be. I can add a `# hash byte size times tree depth` comment instead to clarify the missing bit

Reply:
Ah true - I misget it with signature...